### PR TITLE
Feature/update create ticket modal

### PR DESCRIPTION
**Pull request by gustafvh**
Notion card: 
https://www.notion.so/hedviginsurance/Update-Create-Ticket-modal-design-b4ffa711cf0f4d02ba9f643fa8722be3

- [x]  Default "Include a reminder" as toggled
- [x]  (Prio) Remove `Set priority` and `Current priority` completely
- [x]  (Prio) Remove `Message` , use `Description` = `Message` or vice versa (i.e. they only have to set the Description and the reminder includes the description instead)
- [x]  Default Remind notification date and time to `tomorrow at 9:00` **(Added so it selects the next WEEKday now)**
- [x]  (Optional) Default "Assign to" to self, when applicable

### CODE REVIEW

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 1]
```diff
@@ -67,159 +72,144 @@ export class CreateNewTicket extends React.Component<
...
+                    value={this.state.description}
+                    onChange={(e) => this.handleChange(e)}
+                  />
+                  <Divider />
+                  <Label htmlFor="assignedTo">Assign to:</Label>
+                  <Form.Dropdown
+                    name="assignedTo"
+                    placeholder={
+                      teamOptions[this.state.currentUser.iEXOptionsIndex].text
+                    }
```
Comment (vonElfvin) [inline src/components/tickets/ticket/create-ticket/create-ticket.tsx:161]: Isn't placeholder something different to the default value?
```suggestion
                    defaultValue={ this.state.currentUser.iEXOptionsIndex > -1 ? teamOptions[this.state.currentUser.iEXOptionsIndex].text : null} 
```

File: src/components/tickets/ticket/edit-ticket-mut/reminder.tsx [hunk 2]
```diff
@@ -83,30 +83,8 @@ class ChangeReminderMutation extends React.Component<IChangeReminder, {}> {
...
-                value={this.props.remindMessage}
-                onChange={(e) => this.props.handleChange(e)}
-                maxLength={100}
-                error={
-                  (this.props.touchedDate ||
-                    this.props.touchedTime ||
-                    this.props.touchedMessage) &&
-                  !(this.props.remindMessage.length > 0)
-                }
-              />
```
Comment (vonElfvin) [inline src/components/tickets/ticket/edit-ticket-mut/reminder.tsx:101]: When removing this, please ensure the remind message is still updated if the description is changed.

File: src/features/taskmanager/types.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ export const IEX_TEAM_MEMBERS_OPTIONS: IOption[] = [
   { text: 'Matilda', value: 'matilda@hedvig.com' },
   { text: 'Sebastian', value: 'sebastian@hedvig.com' },
   { text: 'Elvin', value: 'elvin@hedvig.com' },
+  { text: 'Gustaf', value: 'gustaf.halvardsson@hedvig.com' },
```
Comment (vonElfvin) [inline src/features/taskmanager/types.ts:28]: While you are at it you can modify some options:
First `Ebba` should be `Ebba Altefelt`
Then add `Ebba Gustafsson` with email `ebba.gustafsson@hedvig.com`
Add `Claire` with email `claire.mitsell@hedvig.com`

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 4]
```diff
@@ -67,159 +72,144 @@ export class CreateNewTicket extends React.Component<
     description: '',
     setReminder: false,
     type: this.props.type ?? 'REMIND',
+    currentUser: {
+      iEXOptionsIndex: 0,
+      email: null,
+    },
```
Comment (vonElfvin) [inline src/components/tickets/ticket/create-ticket/create-ticket.tsx:78]: I am not sure how the options are set, either this should be 0 as it is, or -1 so the `find` below (which I am pretty sure returns -1 if it is not found) behaves in the same way as the defualt.
```suggestion
    currentUser: {
      iEXOptionsIndex: -1,
      email: null,
    },
```

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 5]
```diff
@@ -67,159 +72,144 @@ export class CreateNewTicket extends React.Component<
...
-              <Form
-                onSubmit={(e) => {
-                  e.preventDefault()
+      <>
+        <Query query={ME}>
+          {({ data }) => {
+            this.state.currentUser.email = data.me
+            this.state.currentUser.iEXOptionsIndex = IEX_TEAM_MEMBERS_OPTIONS.findIndex(
+              (user) => user.value === data.me,
+            )
```
Comment (vonElfvin) [inline src/components/tickets/ticket/create-ticket/create-ticket.tsx:98]: I think this returns -1 if it is not found, check that behaviour for the defaultValue below

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 6]
```diff
@@ -67,159 +72,144 @@ export class CreateNewTicket extends React.Component<
...
+            this.state.assignedTo = data.me
```
[the remark this answers is not in the document]
Comment (gustafvh) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:99]: Accepted in commit, but had to change some syntax

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx (hunk 4 above)
Comment (gustafvh) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:78]: iEXOptionsIndex = 0 defaults to "Unassigned" in iEXOptions which might be a preferable default

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 7]
```diff
@@ -67,159 +72,144 @@ export class CreateNewTicket extends React.Component<
...
+                    <Checkbox
+                      toggle
+                      label="Include a reminder"
+                      checked={this.state.setReminder}
+                      onChange={() => {
+                        const flippedState = !this.state.setReminder
+                        this.setState({
+                          setReminder: flippedState,
+                          remindMessage: this.state.description,
+                        })
```
[the remark this answers is not in the document]
Comment (gustafvh) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:182]: Flipped state is used for toggling whether to display Reminder elements or not so might want to keep that?

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx (hunk 1 above)
Comment (gustafvh) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:161]: defaultValue only populates dropdown field on first render, meaning when getting it from a async query, it populates it with null. placeholder-attribute populates regardless. It stills assigns in the background meaning its only estethic. I could find a workaround

File: src/components/tickets/ticket/edit-ticket-mut/reminder.tsx (hunk 2 above)
Comment (gustafvh) [reply src/components/tickets/ticket/edit-ticket-mut/reminder.tsx:101]: After clean-up, remindMessage wouldnt be used anymore meaning this should be solved automatically

**Comment by gustafvh** [comment]
Added requirements:
- [x]  (Prio) Sort all tickets on how recently they were added
- [x]  (Optional) Add sort ASC/DSC button on creation Date
- [x] Remove all references to priority (except in GraphQL files)

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx [hunk 8]
```diff
@@ -60,170 +64,170 @@ export class CreateNewTicket extends React.Component<
...
-                </Button>
-              </Form>
+                          createNewTicket({
+                            variables: {
+                              ticket: {
+                                assignedTo: this.state.assignedTo,
+                                type: this.state.type,
+                                remindNotificationDate: this.state.remindDate,
+                                remindNotificationTime: this.state.remindTime,
+                                remindMessage: this.state.description,
```
Comment (vonElfvin) [inline src/components/tickets/ticket/create-ticket/create-ticket.tsx:128]: We should probably not include a reminder if it was not toggled to be included

```suggestion
                                remindNotificationDate: this.state.setReminder ? this.state.remindDate : null,
                                remindNotificationTime: this.state.setReminder ? this.state.remindTime : null,
                                remindMessage: this.state.setReminder ? this.state.description : null,
```

Comment (vonElfvin) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:128]: Also, test to ensure that no reminder is sent if "setreminder" is not toggled

File: src/components/tickets/ticket/edit-ticket-mut/reminder.tsx [hunk 9]
```diff
@@ -99,14 +99,8 @@ class ChangeReminderMutation extends React.Component<IChangeReminder, {}> {
                   !(this.props.remindMessage.length > 0)
                 }
               />
-
-              <Button
-                type="submit"
-                compact
-                toggle
-                disabled={this.props.remindMessage.length > 0 ? false : true}
```
Comment (vonElfvin) [inline src/components/tickets/ticket/edit-ticket-mut/reminder.tsx:107]: This should probably be included again

File: src/features/taskmanager/types.ts (hunk 3 above)
Comment (vonElfvin) [reply src/features/taskmanager/types.ts:31]: We also have:
`Joel` with email `joel@hedvig.com`
`Adam` with email `adam.aberg@hedvig.com`

Comment (vonElfvin) [reply src/features/taskmanager/types.ts:31]: I gave wrong info at the first comment, I have now edited it to have the correct information so please double check that.

File: src/components/tickets/ticket/create-ticket/create-ticket.tsx (hunk 8 above)
Comment (gustafvh) [reply src/components/tickets/ticket/create-ticket/create-ticket.tsx:128]: Tested

**Pull request merged**